lib/MusicBrainz/Server/Form/Field/ArtistCredit.pm

```diff
@@ -123,7 +123,14 @@ sub json {
     my $artists = $c->model('Artist')->get_by_ids(map { $_->{artist}->{id} } @$names);
     for my $name (@$names) {
         $name->{artist}->{entityType} = 'artist';
-        $name->{artist}->{gid} = $artists->{$name->{artist}->{id}}->gid if $artists->{$name->{artist}->{id}};
+        my $artist = $artists->{$name->{artist}->{id}};
+        if ($artist)
+        {
+            $name->{artist}->{gid} = $artist->{gid};
+            $name->{artist}->{sortName} = $artist->{sort_name};
+            $name->{artist}->{comment} = $artist->{comment};
+            $name->{artist}->{editsPending} = $artist->{edits_pending};
```

Comment:
Looks good, but there's already an `$artist->TO_JSON` method which should encode things in the same format as here—probably with additional fields that aren't needed, but using it could prevent future issues if the format changes.

As an aside, for blessed Moose objects we always use the accessor methods (`$artist->gid` vs. `$artist->{gid}`). Both will work, but the accessors make it easier to see it's a blessed instance, not a hash reference (and are also easier to type).